File: frontend/src/app/shared/models/character-level.ts
import {Attribute} from './attributes/attribute';
import {AttributeType} from './attributes/attribute-type.enum';

export class CharacterLevel extends Attribute {
  type = 'CharacterLevel';
  minExp = 0;
  profBonus = 0;

  constructor() {
    super();
    this.attributeType = AttributeType.LEVEL;
  }
}

File: backend/src/main/java/com/herd/squire/models/attributes/CharacterLevel.java
package com.herd.squire.models.attributes;

public class CharacterLevel extends Attribute {
    private int minExp;
    private int profBonus;

    public CharacterLevel() { }

    public CharacterLevel(String id, String name, String description, int sid, boolean author, int version, int minExp, int profBonus) {
        super(id, name, description, AttributeType.LEVEL, sid, author, version);
        this.minExp = minExp;
        this.profBonus = profBonus;
    }

    public int getMinExp() {
        return minExp;
    }

    public void setMinExp(int minExp) {
        this.minExp = minExp;
    }

    public int getProfBonus() {
        return profBonus;
    }

    public void setProfBonus(int profBonus) {
        this.profBonus = profBonus;
    }
}
